Chaincase.UI/ViewModels/CoinJoinViewModel.cs

```diff
@@ -0,0 +1,391 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using System.Security;
+using Chaincase;
+using Chaincase.Common;
+using Chaincase.Common.Contracts;
+using Chaincase.Common.Models;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using ReactiveUI;
+using Splat;
+using WalletWasabi.Blockchain.Keys;
+using WalletWasabi.CoinJoin.Client.Clients.Queuing;
+using WalletWasabi.CoinJoin.Client.Rounds;
+using WalletWasabi.CoinJoin.Common.Models;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+
+namespace Chaincase.UI.ViewModels
+{
+	public class CoinJoinViewModel : ReactiveObject
+	{
+		protected Global Global { get; }
+        private CompositeDisposable Disposables { get; set; }
+
+        private INotificationManager notificationManager;
+
+        private string _coordinatorFeePercent;
+		private int _peersRegistered;
+		private int _peersNeeded;
+
+		private RoundPhaseState _roundPhaseState;
+		private DateTimeOffset _roundTimesout;
+		private TimeSpan _timeLeftTillRoundTimeout;
+		private Money _requiredBTC;
+		private Money _amountQueued;
+		private bool _isDequeueBusy;
+		private bool _isEnqueueBusy;
+        private bool _isQueuedToCoinJoin = false;
+		private string _balance;
+        private string _toastErrorMessage;
+        private bool _shouldShowErrorToast;
+        private SelectCoinsViewModel _selectCoinsViewModel;
+        protected StatusViewModel _statusViewModel;
+
+        public CoinJoinViewModel(Global global) 
+        {
+            Global = global;
+            Global.NotificationManager.RequestAuthorization();
+            CoinList = new SelectCoinsViewModel(global);
+            _statusViewModel = new StatusViewModel(global); 
```

Comment:
`StatusViewModel` sould be got from the DI and accessed via constructor parameter. Same with `CoinList`

Reply:
Big bonus would be to move NotificationManager out of global and into the DI container

Reply:
2393bdf5d9f4875c5a71c6e636cb9928ea4ff05b